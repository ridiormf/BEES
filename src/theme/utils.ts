type Space = number | string;

const utils = {
  marginHorizontal: (value: Space) => ({
    marginLeft: value,
    marginRight: value,
  }),
  marginVertical: (value: Space) => ({
    marginTop: value,
    marginBottom: value,
  }),
  paddingHorizontal: (value: Space) => ({
    paddingLeft: value,
    paddingRight: value,
  }),
  paddingVertical: (value: Space) => ({
    paddingTop: value,
    paddingBottom: value,
  }),
};

export default utils;
