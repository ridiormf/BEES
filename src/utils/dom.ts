export const getBody = () => {
  return document.getElementsByTagName('body')[0];
};
