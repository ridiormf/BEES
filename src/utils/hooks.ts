import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

/**
 * Function that simulates the behavior of a class component constructor,
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component is instantiated.
 */
export const useConstructor = <ValueType = void>(
  callback: () => ValueType,
): ValueType | void => {
  const componentWasBuilt = React.useRef<boolean>(false);
  const value = React.useRef<ValueType>();
  if (!componentWasBuilt.current) {
    componentWasBuilt.current = true;
    value.current = callback();
  }

  return value.current;
};

/**
 * Function that simulates the behavior of a class component method "componentDidMount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component mount.
 */
export const useDidMount = (callback: () => void): void => {
  React.useEffect(callback, []);
};

/**
 * Function that simulates the behavior of a class component method "componentWillUnmount",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called as soon as the component will unmount.
 */
export const useWillUnmount = (callback: () => void): void => {
  React.useEffect(() => {
    return callback;
  }, []);
};

/**
 * Function that simulates the behavior of a class component method "componentDidUpdate",
 * adapted to hooks, for functional components.
 *
 * @param {function} callback Callback which will be called on each value change of variables declared as dependencies,
 * excluding the first render of the component.
 * @param {React.DependencyList} deps Dependency list to be listened.
 */
export const useDidUpdate = (
  callback: () => void,
  deps: React.DependencyList,
): void => {
  const firstRender = React.useRef<boolean>(true);
  React.useEffect(() => {
    if (firstRender.current) {
      firstRender.current = false;
    } else {
      callback();
    }
  }, deps);
};

/**
 * Function that use main hooks from "react-router-dom" to simplify usability.
 *
 * @returns {object} containing following attributes:
 * @type {function} "goBack": to return navigation one page.
 * @type {function} "go": to navigate for path.
 * @type {string} "pathname": current url pathname.
 */
export const useAppNavigation = () => {
  const navigate = useNavigate();
  const location = useLocation();

  const goBack = () => {
    navigate(-1);
  };

  const go = (path: string) => {
    navigate(path);
  };

  return {
    goBack,
    go,
    pathname: location.pathname,
  };
};
