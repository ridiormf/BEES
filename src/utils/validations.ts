/**
 * Function that checks if the full name is valid.
 * It will check:
 * 1. if the string has, at least, two names, and if all names has 2 letters or more;
 * 2. If the string not start or end with blank space;
 * 3. If it doesn't contain numbers; and
 * 4. If it doesn't contain two or more blank spaces between words
 *
 * @param {string} fullName String to be checked.
 * @returns {boolean} Return true if is a valid name, otherwise returns false.
 */
export const checkFullName = (fullName?: string): boolean => {
  if (!fullName) return false;

  const splittedNames = fullName.split(' ');

  return (
    splittedNames.length > 1 &&
    splittedNames.every((name) => name.length > 1 && !/\d/g.test(name))
  );
};
