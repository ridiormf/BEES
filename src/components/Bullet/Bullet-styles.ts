import theme from '../../theme';
import ComponentText from '../Text';

const Bullet = theme.styled('button', {
  paddingVertical: 0,
  paddingHorizontal: 16,
  lineHeight: 0,
  fontSize: 16,
  minHeight: 38,
  border: 'none',
  borderRadius: 200,
  backgroundColor: '$primaryBackground',
  transition: '200ms',
  display: 'flex',
  alignItems: 'center',
  gap: 8,
  '&:not([disabled]):hover': {
    cursor: 'pointer',
    filter: 'brightness(1.05)',
  },
  '&:not([disabled]):active': {
    filter: 'brightness(1.1)',
  },
});

const Text = theme.styled(ComponentText, {
  fontSize: 16,
  lineHeight: '26px',
  color: '$secondaryText',
  marginVertical: 4,
  overflow: 'hidden',
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
});

export const bulletStyles = {
  styled: {
    Bullet,
    Text,
  },
};
