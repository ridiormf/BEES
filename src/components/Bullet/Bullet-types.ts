import React from 'react';
import { bulletStyles } from './Bullet-styles';

const {
  styled: { Bullet },
} = bulletStyles;

export interface BulletProps
  extends Omit<React.ComponentProps<typeof Bullet>, 'onChange'> {
  children: React.ReactNode;
  icon?: string;
  imageAlt?: string;
}
