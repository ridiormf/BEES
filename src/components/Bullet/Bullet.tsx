import React from 'react';
import Image from '../Image';

import { bulletStyles } from './Bullet-styles';
import { BulletProps } from './Bullet-types';

const {
  styled: { Bullet: StyledBullet, Text },
} = bulletStyles;

const Bullet: React.FC<BulletProps> = ({
  children,
  icon,
  imageAlt,
  onClick,
  ...rest
}) => {
  return (
    <StyledBullet {...rest} onClick={onClick} disabled={!onClick}>
      {icon && imageAlt ? <Image src={icon} alt={imageAlt} /> : undefined}
      <Text>{children}</Text>
    </StyledBullet>
  );
};

export default Bullet;
