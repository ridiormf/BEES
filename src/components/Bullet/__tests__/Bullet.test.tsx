import { render } from '@testing-library/react';
import { Wrapper } from '../../../__tests__/utils';
import Bullet from '../Bullet';

describe('components ->', () => {
  describe('Bullet ->', () => {
    describe('snapshots ->', () => {
      it('Should match snapshot', () => {
        const { container } = render(
          <Bullet>
            <></>
          </Bullet>,
          { wrapper: Wrapper },
        );
        expect(container).toMatchSnapshot();
      });
    });
  });
});
