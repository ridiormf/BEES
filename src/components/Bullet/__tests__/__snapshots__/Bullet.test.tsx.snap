// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components -> Bullet -> snapshots -> Should match snapshot 1`] = `
<div>
  <button
    class="c-bETMrI"
    disabled=""
  >
    <p
      class="c-caxBfD c-ljOCux"
    />
  </button>
</div>
`;
